### PR TITLE
[DD4hep] Resolve static analyzer warning by protecting against divide by 0

### PR DESCRIPTION
**Pull request by cvuosalo**
Issue #31288 lists static analyzer warnings that needed to be fixed. This PR fixes the last one by protecting against division by 0.

Note that the method that was fixed appears to be no longer used. Also, the class for this method appears to be used only for a test, so none of this code is used in production.

#### PR validation:

Another test program was altered to call the code changed by this PR to verify that the fix works correctly.

No backport is needed.

### CODE REVIEW

**Comment by cvuosalo** [comment]
@cmsbuild please test

**Comment by civanch** [comment]
+1

File: Geometry/MuonNumbering/src/DD4hep_MuonNumbering.cc [hunk 1]
```diff
@@ -18,7 +18,11 @@ const MuonBaseNumber MuonNumbering::geoHistoryToBaseNumber(const cms::ExpandedNo
     edm::LogError("Geometry") << "MuonNumbering finds unusual base constant:" << basePart;
   }
   if (superPart < 100) {
-    edm::LogError("Geometry") << "MuonNumbering finds unusual super constant:" << superPart;
+    edm::LogError("Geometry") << "MuonNumbering finds unusual super part:" << superPart;
+    if (superPart <= 0) {
+      edm::LogError("Geometry") << "MuonNumbering -- resetting super part to 100";
+      superPart = 100;  // Protect against division by 0
```
Comment (perrotta) [inline Geometry/MuonNumbering/src/DD4hep_MuonNumbering.cc:24]: I imagine that this is just hypothetical, and only made in order to quiet down the static analyzer: but why "100", and not (e.g.) "1"? If superPart is 1 or 2 it remains as such, if it is 0 it goes to 100...

Comment (civanch) [reply Geometry/MuonNumbering/src/DD4hep_MuonNumbering.cc:24]: I would agree with Andrea - the check (<0) is problematic. May be a correct code would be without this "if" - all superPart should be above 100 or 100.

Comment (cvuosalo) [reply Geometry/MuonNumbering/src/DD4hep_MuonNumbering.cc:24]: Yes, I'll drop the `<=0` check and combine the two messages.

**Comment by cvuosalo** [comment]
hold

**Comment by cvuosalo** [comment]
A revision to this PR is on the way.

**Comment by cvuosalo** [comment]
unhold

**Comment by cvuosalo** [comment]
@cmsbuild please test

**Comment by cvuosalo** [comment]
+1

**Comment by qliphy** [comment]
+1
